src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/SqlAction.cs

```diff
@@ -648,6 +649,125 @@ TraceId traceId
         return result;
     }
 
+    public async Task<CascadingUpdateResult?> CascadeUpdates(
+        string resourceName,
+        long documentId,
+        short documentPartitionKey,
+        DocumentInfo documentInfo,
+        NpgsqlConnection connection,
+        NpgsqlTransaction transaction,
+        TraceId traceId
+    )
+    {
+        var result = await GetPostgresExceptionRetryPipeline()
+            .ExecuteAsync(async cancellationToken =>
+            {
+                try
+                {
+                    string BuildUpdateStatement(int n)
+                    {
+                        string retStr = $"jsonb_set(";
+                        string resourceReference =
+                            $"{char.ToLowerInvariant(resourceName[0]) + resourceName[1..]}Reference";
```

Comment:
String manipulation like this is always an anti-pattern. ODS/API naming conventions are unfortunately far too weird for this to work.

Reply:
Don't forget: there are additional tickets around references and role-named entities. Hopefully we can make tangible progress with this ticket.